Check for file existence in packageFiles instead of matching strings

d-torrance opened the pull request:
This is much more robust, and also avoids an issue where `realpath` was stripping away the filename so that the "startup.m2" regex didn't match.

In particular, this was the big problem:

```m2
i1 : User#"source file"

o1 = /usr/local/share/Macaulay2/Core/startup.m2

i2 : realpath oo

o2 = /usr/local/share/Macaulay2
```

I'm not sure if this is the expected behavior of `realpath` or not, but that might be a bug too.

d-torrance committed:
Check for file existence in packageFiles instead of matching strings

This is much more robust, and also avoids an issue where "realpath"
was stripping away the filename so that the "startup.m2" regex didn't
match.

mahrud reviewed (approved):
This is good, but I'd rather fix `User#"source file"` to something more reasonable, like stdio.

mahrud commented:
This is still not good:
```m2
ii10 : debug Core

ii11 : packageFiles pkg

oo11 = {/home/mahrud/Projects/M2/M2/M2/Macaulay2/m2/startup.m2.in}

oo11 : List

ii12 : file

oo12 = /home/mahrud/Projects/M2/M2/M2/Macaulay2/m2/startup.m2.in

ii13 : filesLoaded#file
stdio:13:11:(3): error: key not found in hash table:
        "/home/mahrud/Projects/M2/M2/M2/Macaulay2. (of class String)

ii14 : filesLoaded#?file

oo14 = false
```

I think we need one or both of these changes:
- `needsPackage` should be nilpotent for User, Core, etc.
- source file for User needs to be stdio or currentString

d-torrance committed:
Set User source file to stdio

It was previously startup.m2.in, even though it's actually defined in
last.m2.  But stdio makes sense since that's where the user will
actually define the package's symbols.
